R/query_index.R

```diff
@@ -63,15 +70,43 @@ query_index <- R6::R6Class(
     #'
     #' @return A new query_index object with the scoped index
     get_index_scoped = function() {
-      query_index$new(self$root, private$index_scoped)
+      query_index$new(self$root, private$index_scoped, private$depends)
+    },
+
+    #' @description
+    #' Get the ids of packets which this packet depends on i.e. its parents
+    #' this can get immediate parents or recurse to build whole tree
+    #'
+    #' @param id The id of the packet to get parents of
+    #' @param immediate If TRUE get just immediate (level 1) parents
+    #' @return The ids of the parents of this packet
+    get_packet_depends = function(id, immediate) {
+      if (immediate) {
+        deps <- self$depends[[id]]$packet
+      } else {
+        deps <- unique(private$get_all_packet_depends(id))
+      }
+      deps <- deps[deps %in% self$index$id] ## filter on current index
```

Comment:
Is this not

```suggestion
      deps <- intersect(deps, self$index$id)
```

which is them simple enough to write

```suggestion
      intersect(deps, self$index$id) %||% character(0)
```